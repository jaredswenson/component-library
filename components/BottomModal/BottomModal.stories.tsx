import React from "react";
import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react-native";
import BottomModal from "./BottomModal";
import Text from '../Text/Text'

export default {
  title: "Bottom Modal",
  component: BottomModal,
};

// Stories
export const Modal = () => {
  return(
    <BottomModal modal={true}>
      <Text text="Modal"/>
    </BottomModal>
  )
};

// Add all stories to RN/Expo storybook
storiesOf("Modal", module).add("Modal", Modal);
