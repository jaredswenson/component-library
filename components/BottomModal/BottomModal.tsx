import React from 'react';
import { Modal, StyleSheet } from 'react-native';
import { BlurView } from 'expo-blur';

interface BottomModalProps {
  modal: boolean;
  blur?: number;
}

const BottomModal: React.FC<BottomModalProps>  = ({ children, modal, blur = 100 }) => {
  return (
    <Modal animationType="slide" visible={modal} transparent>
      <BlurView
        intensity={blur}
        tint="dark"
        style={[StyleSheet.absoluteFill, { backgroundColor: '#000' }]}
      >
        {children}
      </BlurView>
    </Modal>
  );
};

export default BottomModal;