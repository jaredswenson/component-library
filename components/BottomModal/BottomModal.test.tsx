import React from "react";
import renderer from "react-test-renderer";

import BottomModal from "./BottomModal";

describe("<BottomModal />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<BottomModal modal={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
