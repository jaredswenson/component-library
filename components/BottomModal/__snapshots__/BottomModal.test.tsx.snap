// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomModal /> renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "#000",
          },
        ],
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
  >
    <ViewManagerAdapter_ExpoBlurView
      proxiedProperties={
        Object {
          "intensity": 100,
          "tint": "dark",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</Modal>
`;
